modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -103,6 +109,11 @@ func (h *handler) DeployBundle(bd *fleet.BundleDeployment, status fleet.BundleDe
 	if err != nil {
 		return status, err
 	}
+
+	if status.LastApply == nil {
```

Comment:
Does this only get set once? If it is not nil then it will not be set. Also we might want to set it when it actuallu deploy bundle, since `h.deployManager.Deploy(bd)` could be a no-op.

Reply:
changed the logic to use different in releaseName to identify if a new release has been performed and use the same to calculate LastApply timestamp.